ado_asana_sync/sync/asana.py

```diff
@@ -26,32 +28,39 @@ def get_asana_task(app: App, task_gid: str) -> object | None:
     :return: Task object or None if no task is found.
     :rtype: object or None
     """
-    api_instance = asana.TasksApi(app.asana_client)
-    try:
-        # Get all tasks in the project.
-        opt_fields = [
-            "assignee_section",
-            "due_at",
-            "name",
-            "completed_at",
-            "tags",
-            "dependents",
-            "projects",
-            "completed",
-            "permalink_url",
-            "parent",
-            "assignee",
-            "assignee_status",
-            "num_subtasks",
-            "modified_at",
-            "workspace",
-            "due_on",
-        ]
-        api_response = api_instance.get_task(
-            task_gid,
-            opt_fields=opt_fields,
+    with _TRACER.start_as_current_span("get_asana_task") as span:
+        span.set_attributes(
+            {
+                "asana_workspace_name": app.asana_workspace_name,
+                "task_gid": task_gid,
+            }
         )
-        return api_response.data
-    except ApiException as exception:
-        _LOGGER.error("Exception when calling TasksApi->get_task: %s\n", exception)
-        return None
+        api_instance = asana.TasksApi(app.asana_client)
+        try:
+            # Get all tasks in the project.
+            opt_fields = [
+                "assignee_section",
+                "due_at",
+                "name",
+                "completed_at",
+                "tags",
+                "dependents",
+                "projects",
+                "completed",
+                "permalink_url",
+                "parent",
+                "assignee",
+                "assignee_status",
+                "num_subtasks",
+                "modified_at",
+                "workspace",
+                "due_on",
+            ]
+            api_response = api_instance.get_task(
+                task_gid,
+                opt_fields=opt_fields,
+            )
+            return api_response.data
+        except ApiException as exception:
+            _LOGGER.error("Exception when calling TasksApi->get_task: %s\n", exception)
+            return None
```

Comment:
**Suggestion:** The `get_asana_task` method is quite large and complex. Consider breaking it down into smaller, more manageable methods. This would make the code easier to read and maintain.
```suggestion
    # Suggestion: Break down the `get_asana_task` method into smaller methods
```